Composer/packages/client/src/constants.ts

```diff
@@ -210,6 +210,8 @@ export const QnABotTemplateId = 'QnASample';
 
 export const nameRegex = /^[a-zA-Z0-9-_]+$/;
 
+export const dialogNameRegex = /^[a-zA-Z][a-zA-Z0-9-_]*$/;
```

Comment:
Instead of maintain another regex, can we make `nameRegex` more strict to meet this requirements?

Reply:
I remember in server also have a `nameRegex`

Reply:
Thanks @zhixzhan , I moves this to shared, and updated the server check. The nameRegex is also used by intent and trigger. I not sure if we can update the regex.